pdfminer/pdfpage.py

```diff
@@ -68,7 +68,8 @@ def __init__(
             self.attrs.get("Resources", dict()),
         )
         mediabox_params: List[Any] = [
-            resolve1(mediabox_param) for mediabox_param in self.attrs["MediaBox"]
+            resolve1(mediabox_param)
+            for mediabox_param in resolve1(self.attrs["MediaBox"])
```

Comment:
We sometimes hit `KeyError` in this line, e.g. `self.attrs`
```
{
    Parent: <PDFObjRef:2>,
    Type: /'Page'
}
```
Can we protect it against missing `MediaBox`?
```python
mediabox_params: List[Any] = []
if "MediaBox" in self.attrs:
    mediabox_params = [
        resolve1(mediabox_param)
        for mediabox_param in resolve1(self.attrs["MediaBox"])
    ]
```

Reply:
Indeed... MediaBox is required by the spec (at least for PDF 1.7) but as usual this doesn't mean that it will necessarily be present in random PDFs.

Reply:
Ah, I misspoke slightly - MediaBox is inheritable, so it isn't required on individual `Page`s, and `pdfminer` should always propagate it from the top-level `Pages` object.  If it's missing there, then we have bigger problems, do you have a test case where this is happening?

Reply:
> do you have a test case where this is happening?

Unfortunately, I cannot share the PDF file :disappointed:

Reply:
No problem, in any case the code in this PR should be robust to `MediaBox` missing everywhere.  It is suprising that you would have a `Page` that fails to inherit it though, which might indicate some other bug in `pdfminer`.